### PR TITLE
Fix sign in and reserve

### PR DESCRIPTION
**Pull request by vincit-matu**

### CODE REVIEW

File: apps/ui/components/calendar/ReservationCalendarControls.tsx [hunk 1]
```diff
@@ -56,6 +56,8 @@ import {
 import LoginFragment from "../LoginFragment";
 import { useDebounce } from "@/hooks/useDebounce";
 import { capitalize, formatDurationMinutes } from "@/modules/util";
+import { NEXT_URL } from "next/dist/client/components/app-router-headers";
```
Comment (joonatank) [inline apps/ui/components/calendar/ReservationCalendarControls.tsx:59]: lint...

File: apps/ui/components/calendar/ReservationCalendarControls.tsx [hunk 2]
```diff
@@ -603,6 +605,9 @@ const ReservationCalendarControls = <T extends Record<string, unknown>>({
     (n) => n?.endDatetime
   );
 
+  let loc;
+  if (isBrowser) loc = window.location;
```
Comment (joonatank) [inline apps/ui/components/calendar/ReservationCalendarControls.tsx:609]: not a fan of ternaries, but let on line 600 of a 2000 line function is worse.
`const loc = isBrowser ? window.location : undefined;`

File: apps/ui/hooks/auth/index.ts [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
-export function signIn() {
-  const currentUrl = window.location.href;
-  const url = getSignInUrl(currentUrl);
+// Redirect the user to the sign in dialog and return to returnUrl (or
+// current url if none is provided) after sign in
+export function signIn(returnUrl?: string) {
+  const returnTo = returnUrl ?? window.location.href;
+  const url = getSignInUrl(returnTo);
   window.location.href = url;
 }
```
Comment (joonatank) [inline apps/ui/hooks/auth/index.ts:10]: This breaks the Login button. (not sure why)
Should in any case check for invalid urls and empty strings.

File: apps/ui/components/calendar/ReservationCalendarControls.tsx (hunk 1 above)
Comment (vincit-matu) [reply apps/ui/components/calendar/ReservationCalendarControls.tsx]: fixed

File: apps/ui/components/calendar/ReservationCalendarControls.tsx [hunk 4]
```diff
@@ -634,6 +637,13 @@ const ReservationCalendarControls = <T extends Record<string, unknown>>({
             {t("reservationCalendar:makeReservation")}
           </SubmitButton>
         }
+        returnUrl={
+          loc
+            ? `${loc.protocol}//${loc.hostname}${
+                loc.port ? `:${loc.port}` : ""
+              }/reservation-unit/${reservationUnit.pk}?isPostLogin=true`
+            : undefined
```
[the remark this answers is not in the document]
Comment (vincit-matu) [reply apps/ui/components/calendar/ReservationCalendarControls.tsx:645]: This is probably not necessary in production. But I had problems testing locally if I didn't construct the url.. 😮

File: apps/ui/pages/reservation-unit/[id].tsx [hunk 5]
```diff
@@ -432,6 +433,7 @@ const ReservationUnit = ({
   relatedReservationUnits,
   activeApplicationRounds,
   termsOfUse,
+  isPostLogin,
```
[the remark this answers is not in the document]
Comment (vincit-matu) [reply apps/ui/pages/reservation-unit/[id].tsx:436]: It's defined in the getServeSideProps return statement.

File: apps/ui/components/calendar/ReservationCalendarControls.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (joonatank) [reply apps/ui/components/calendar/ReservationCalendarControls.tsx:645]: Like https://github.com/City-of-Helsinki/tilavarauspalvelu-ui/blob/6b972601876acecbe8d9650b4abea1ef4daa1d0e/apps/ui/modules/reservation.ts#L412

File: apps/ui/components/calendar/ReservationCalendarControls.tsx [hunk 6]
```diff
@@ -634,6 +645,7 @@ const ReservationCalendarControls = <T extends Record<string, unknown>>({
             {t("reservationCalendar:makeReservation")}
           </SubmitButton>
         }
+        returnUrl={() => getPostLoginUrl()}
```
Comment (joonatank) [inline apps/ui/components/calendar/ReservationCalendarControls.tsx:648]: ` returnUrl={getPostLoginUrl()}`?

**Review by joonatank — APPROVED** [review]
🚀  squash first

**Pull request merged**